block-node/block-providers/files.recent/src/test/java/org/hiero/block/node/blocks/files/recent/BlockFileBlockAccessorTest.java

```diff
@@ -232,12 +235,71 @@ void testBlock(final CompressionType compressionType) throws IOException {
             final Block expected = new Block(List.of(blockItems));
             final Bytes protoBytes = Block.PROTOBUF.toBytes(expected);
             // create instance to test
-            final BlockFileBlockAccessor toTest = createInstanceToTest(blockFilePath, compressionType, protoBytes);
+            final BlockFileBlockAccessor toTest =
+                    createBlockAndGetAssociatedAccessor(blockFilePath, compressionType, protoBytes);
             // test accessor.block()
             final Block actual = toTest.block();
             assertThat(actual).isEqualTo(expected);
         }
 
+        /**
+         * This test aims to verify that the {@link BlockFileBlockAccessor#block()} will correctly handle
+         * IOExceptions encountered when attempting to persist blocks.
+         */
+        @ParameterizedTest
+        @EnumSource(CompressionType.class)
+        @DisplayName("Test block() method correctly handles an IOException")
+        void testBlockIOException(final CompressionType compressionType) throws IOException {
+            // create block file path before call
+            final Path blockFilePath = testBasePath.resolve("0.blk".concat(compressionType.extension()));
+            // build a test block
+            final BlockItem[] blockItems = SimpleTestBlockItemBuilder.createNumberOfVerySimpleBlocks(1);
+            final Block expected = new Block(List.of(blockItems));
+            final Bytes protoBytes = Block.PROTOBUF.toBytes(expected);
+            // create instance to test
+            final BlockFileBlockAccessor toTest =
+                    createBlockAndGetAssociatedAccessor(blockFilePath, compressionType, protoBytes);
+
+            // delete the file to simulate NoSuchFileException IOException
+            Files.delete(blockFilePath);
+
+            // test accessor.block() expecting IOException
+            try {
+                toTest.block();
+                assertThat(false).isTrue(); // exception should have been thrown
+            } catch (Throwable e) {
+                // expected
+                Assertions.assertThat(e).isInstanceOf(UncheckedIOException.class);
+            }
```

Comment:
AssertJ offers fluent apis for checking exceptions thrown. The below code should be in place of the whole try-catch. It will fail if no exception is thrown.  You can consider this approach to be applied everywhere.

```suggestion
            assertThatExceptionOfType(UncheckedIOException.class).isThrownBy(toTest::block);
```

On another note, if for any reason you want to fail the test, we should use `Assertions.fail("some message");` (again coming from AssertJ) instead of trying to produce a fake assert. But in this case none of this should be used.

Reply:
Agreed, thanks. Will adopt